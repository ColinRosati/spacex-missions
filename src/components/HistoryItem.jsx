import React from 'react';
import '../styles/history.css';

function HistoryItem(props) {
  const {title, details, event_date_utc,links} = props
  return (
    <div className="history-item">
      <div className="history-item-title">{title}</div>
      <div className="history-item-details">{details}</div>
      <div className="history-item-event">{event_date_utc}</div>
      <div className="history-item-link"><a href={links}>more</a></div>
    </div>
  );
}

export default HistoryItem;
