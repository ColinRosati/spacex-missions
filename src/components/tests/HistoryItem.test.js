import HistoryItem from '../HistoryItem';
import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

// test # of props
// test length of div children


Enzyme.configure({ adapter: new Adapter() }) // look up how to do this globally

describe('historyItem component', () => {
    it('should render hitoryitems component', () => {
        const wrapper = shallow(
            <div className="history-item">
                <div className="history-item-title"></div>
                <div className="history-item-details"></div>
                <div className="history-item-event"></div>
                <div className="history-item-link"><a href>more</a></div>
            </div>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should expect props to be valid', () => {
        const it = mount(<HistoryItem/>);
        expect(it.props().title).not.toBeNull();
        expect(it.props().details).not.toBeNull();
        expect(it.props().event_date_utc).not.toBeNull();
        expect(it.props().links).not.toBeNull();
    })

    it('should have 5 divs', () => {
        const wrapper = shallow( <HistoryItem/> )
        const children = wrapper.find('div').children()
        console.log(children.length)
        expect(children.length).toBe(5);
    })

    
    it('should be show contain text', () => {
        const wrapper = shallow( <HistoryItem/> )
        const text = wrapper.find('.history-item-link')
        expect(text.text()).toBe('more')
    })


})