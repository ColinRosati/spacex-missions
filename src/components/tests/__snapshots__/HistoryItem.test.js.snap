// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`historyItem component should render hitoryitems component 1`] = `ShallowWrapper {}`;
